cmd/podmanV2/containers/logs.go

```diff
@@ -0,0 +1,110 @@
+package containers
+
+import (
+	"os"
+
+	"github.com/containers/libpod/cmd/podmanV2/registry"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/containers/libpod/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// logsOptionsWrapper wraps entities.LogsOptions and prevents leaking
+// CLI-only fields into the API types.
+type logsOptionsWrapper struct {
+	entities.ContainerLogsOptions
+
+	SinceRaw string
+}
+
+var (
+	logsOptions     logsOptionsWrapper
+	logsDescription = `Retrieves logs for one or more containers.
+
+  This does not guarantee execution order when combined with podman run (i.e., your run may not have generated any logs at the time you execute podman logs).
+`
+	logsCommand = &cobra.Command{
+		Use:     "logs [flags] CONTAINER [CONTAINER...]",
+		Short:   "Fetch the logs of one or more container",
+		Long:    logsDescription,
+		RunE:    logs,
+		PreRunE: preRunE,
+		Example: `podman logs ctrID
+  podman logs --names ctrID1 ctrID2
+  podman logs --tail 2 mywebserver
+  podman logs --follow=true --since 10m ctrID
+  podman logs mywebserver mydbserver`,
+	}
+
+	containerLogsCommand = &cobra.Command{
+		Use:     logsCommand.Use,
+		Short:   logsCommand.Short,
+		Long:    logsCommand.Long,
+		PreRunE: logsCommand.PreRunE,
+		RunE:    logsCommand.RunE,
+		Example: `podman container logs ctrID
+		podman container logs --names ctrID1 ctrID2
+		podman container logs --tail 2 mywebserver
+		podman container logs --follow=true --since 10m ctrID
+		podman container logs mywebserver mydbserver`,
+	}
+)
+
+func init() {
+	// logs
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode},
+		Command: logsCommand,
+	})
+
+	logsCommand.SetHelpTemplate(registry.HelpTemplate())
+	logsCommand.SetUsageTemplate(registry.UsageTemplate())
+
+	flags := logsCommand.Flags()
+	logsFlags(flags)
+
+	// container logs
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode},
+		Command: containerLogsCommand,
+		Parent:  containerCmd,
+	})
+
+	containerLogsCommand.SetHelpTemplate(registry.HelpTemplate())
+	containerLogsCommand.SetUsageTemplate(registry.UsageTemplate())
+
+	containerLogsFlags := containerLogsCommand.Flags()
+	logsFlags(containerLogsFlags)
+}
+
+func logsFlags(flags *pflag.FlagSet) {
+	flags.BoolVar(&logsOptions.Details, "details", false, "Show extra details provided to the logs")
```

Comment:
For some reason Podmna logs has

	flags.SetInterspersed(false)

Reply:
I added it back but I think that's not needed for logs. All arguments must be container names or IDs. But one way or another, we'll error out if flag-like args are passed that we don't know.